diff --git a/helix-term/src/ui/editor.rs b/helix-term/src/ui/editor.rs
index afc88ef..efadf10 100644
--- a/helix-term/src/ui/editor.rs
+++ b/helix-term/src/ui/editor.rs
@@ -593,7 +593,6 @@ pub fn tabstop_highlights(doc: &Document, theme: &Theme) -> Option<OverlayHighli
 
     /// Render bufferline at the top
     pub fn render_bufferline(editor: &Editor, viewport: Rect, surface: &mut Surface) {
-        let scratch = PathBuf::from(SCRATCH_BUFFER_NAME); // default filename to use for scratch buffer
         surface.clear_with(
             viewport,
             editor
@@ -616,13 +615,15 @@ pub fn render_bufferline(editor: &Editor, viewport: Rect, surface: &mut Surface)
         let current_doc = view!(editor).doc;
 
         for doc in editor.documents() {
-            let fname = doc
-                .path()
-                .unwrap_or(&scratch)
-                .file_name()
-                .unwrap_or_default()
-                .to_str()
-                .unwrap_or_default();
+            let fname = if let Some(path) = doc.path() {
+                path.file_name()
+                    .and_then(|s| s.to_str())
+                    .unwrap_or_default()
+            } else if let Some(name) = doc.name.as_deref() {
+                name
+            } else {
+                SCRATCH_BUFFER_NAME
+            };
 
             let style = if current_doc == doc.id() {
                 bufferline_active
